test/test_torch.py

```diff
@@ -153,6 +155,53 @@ def make_contiguous_slice(size, dtype):
 
         return tensors
 
+    @unittest.skipIf(not TEST_CUDA, 'CUDA not available')
+    @skipIfRocm
+    def test_cublas_multiple_threads_same_device(self):
```

Comment:
After offline discussion with @mcarilli, I added this unit test. On my local machine, this test consistently fails previously but after this PR it passes consistently. This PR is now complete and ready for review.

Reply:
Why did you put the test at this location?  My old `test_cudnn_multiple_threads_same_device` is in `test_nn.py`, among the other convolution tests.  I think you should either put this near the other cublas tests, or decide on some new section dedicated to handles, and put these two tests next to each other. If they're next to each other, you could also make some logic generic/avoid some code duplication between them (the only difference is that one calls a convolution and the other calls a gemm).

Reply:
Both can go into test_cuda.py